packages/devtools_app/lib/src/timeline/flutter/timeline_flame_chart.dart

```diff
@@ -78,7 +87,7 @@ class FrameBasedTimelineFlameChart extends StatefulWidget {
 
   final double height;
 
-  final TimelineEvent Function() selectionProvider;
+  final ValueNotifier<TimelineEvent> selectionNotifier;
```

Comment:
Nit: create private ValueNotifier fields and expose only the ValueListenable interface as the public api.
That way users of the controller are clear that they should not modify the value on the controller and should instead only listen for changes to the value on the controller.
h

Reply:
done